libtransact/src/contract/context/key_value.rs

```diff
@@ -0,0 +1,1310 @@
+    ///
+    /// * `entries` - Hashmap with the map to be stored in state at the associated natural key
+    pub fn set_state_entries(
+        &self,
+        entries: HashMap<&K, HashMap<String, ValueType>>,
+    ) -> Result<(), KeyValueContextError> {
+        let keys = entries.keys().map(ToOwned::to_owned).collect::<Vec<&K>>();
+        let addresses = keys
+            .iter()
+            .map(|k| {
+                self.addresser
+                    .compute(&k)
+                    .map_err(KeyValueContextError::AddresserError)
+            })
+            .collect::<Result<Vec<String>, KeyValueContextError>>()?;
+        // Creating a map of the StateEntryList to the address it is stored at
+        let entry_list_map = self.get_state_entry_lists(&addresses)?;
+
+        // Iterating over the provided HashMap to see if there is an existing StateEntryList at the
+        // corresponding address. If there is one found, add the new StateEntry to the StateEntryList.
+        // If there is none found, creates a new StateEntryList entry for that address.
+        // Then, serializes the newly created StateEntryList to be set in the internal context.
+        let entries_list = entries
+            .iter()
+            .map(|(key, values)| {
+                let addr = self.addresser.compute(key)?;
+                let state_entry = self.create_state_entry(key, values.to_owned())?;
+                match entry_list_map.get(&addr) {
+                    Some(entry_list) => {
+                        let mut existing_entries = entry_list.entries().to_vec();
+                        existing_entries.push(state_entry);
+                        let entry_list = StateEntryListBuilder::new()
+                            .with_state_entries(existing_entries)
+                            .build()
+                            .map_err(|err| {
+                                KeyValueContextError::ProtocolBuildError(Box::new(err))
+                            })?;
+                        Ok((addr, entry_list.into_bytes()?))
+                    }
+                    None => {
+                        let entry_list = StateEntryListBuilder::new()
+                            .with_state_entries(vec![state_entry])
+                            .build()
+                            .map_err(|err| {
+                                KeyValueContextError::ProtocolBuildError(Box::new(err))
+                            })?;
+                        Ok((addr, entry_list.into_bytes()?))
+                    }
+                }
+            })
+            .collect::<Result<Vec<(String, Vec<u8>)>, KeyValueContextError>>()?;
+        self.context.set_state_entries(entries_list)?;
+
+        Ok(())
+    }
+
+    /// get_state_entries attempts to retrieve the data from state at the addresses calculated from
+    /// the provided list of natural keys.
+    ///
+    /// Returns a HashMap containing the normalized key associated with a HashMap, representing the
```

Comment:
Using "a HashMap" twice in this sentence is confusing. Are two different (or potentially different) HashMaps being discussed? If not, perhaps the second one should be "the HashMap"? 

Also, it's not clear (semantically) what represents the data retrieved from state: the HashMap or the normalized key.  Can you clarify?

Reply:
Reworded for clarity